### PR TITLE
Fixing Automation Tests for SDK based on Swagger 

### PR DESCRIPTION
**Pull request by vrdmr**
## Description

Generating new SDK based on Swagger. Fixed the tests and added two new resources in tests. Other resources are being added as well.
- Rest spec link - https://github.com/Azure/azure-rest-api-specs/commit/ed7a298def311fe2c3265e30293639fcf4355b87
- Bumping to 3.0.0-preview as it contains breaking changes. We'll be keeping it preview for a while, till all our cmdlets and internal test suites are moved to newer SDK.
- It contains newer recorded tests and tested multiple times.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by vrdmr** [comment]
@safeermohammed @Francisco-Gamino @weijiel @Diastro - FYI

File: src/SDKs/Automation/Automation.Tests/Customizations/AutomationManagementClient.Customization.cs [hunk 1]
```diff
@@ -3,12 +3,11 @@
...
 using Microsoft.WindowsAzure.Management.Automation;
 
-namespace Microsoft.WindowsAzure.Management.Automation
+namespace Automation.Tests.Customizations
 {
-    public partial class AutomationManagementClient
+    public class AutomationManagementClient
     {
         public static List<T> ContinuationTokenHandler<T>(Func<string, ResponseWithSkipToken<T>> listFunc)
         {
```
Comment (safeermohammed) [inline src/SDKs/Automation/Automation.Tests/Customizations/AutomationManagementClient.Customization.cs]: Do we need this? I do not think we have an SkipTokens
only skip and take

File: src/SDKs/Automation/Automation.Tests/Helpers/ResourceGroupHelper.cs [hunk 2]
```diff
@@ -4,27 +4,25 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using Microsoft.Azure.Management.Automation;
+using Microsoft.Azure.Management.Automation.Testing;
 using Microsoft.Azure.Management.Resources;
-//using Microsoft.Azure.Test;
 using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
-using System.Net.Http;
+//using Microsoft.Azure.Test;
```
Comment (safeermohammed) [inline src/SDKs/Automation/Automation.Tests/Helpers/ResourceGroupHelper.cs]: remove

**Comment by shahabhijeet** [comment]
@vrdmr add to whitelist

**Comment by shahabhijeet** [comment]
@vrdmr please log in to Travis, so that your user is registered in Travis. This will cause travis to kick off the CI job.

**Comment by shahabhijeet** [comment]
@vrdmr you need to log in to travis-ci.com and register your GitHub id.
Once you do that, make a commit to your PR and that will kick off the travis build.

**Comment by vrdmr** [comment]
@shahabhijeet Done. I'll make the commit with the changes requested by @safeermohammed and kick of the travis build. Thanks.

File: src/SDKs/Automation/Automation.Tests/Customizations/AutomationManagementClient.Customization.cs (hunk 1 above)
Comment (vrdmr) [reply src/SDKs/Automation/Automation.Tests/Customizations/AutomationManagementClient.Customization.cs]: I will remove the Customizations folder. We do not need this now.

**Comment by vrdmr** [comment]
@shahabhijeet - I see the following: https://travis-ci.org/Azure/azure-sdk-for-net/requests with the "Abuse Detected" but nothing else to tell what exactly was the issue.

```
#3876 e4934c3
 8 minutes ago Merge remote-tracking branch 'upstream/psSdkJson6' into psSdkJson6 - Abuse detected
```

File: src/SDKs/Automation/Automation.Tests/Automation.Tests.csproj [hunk 3]
```diff
@@ -1,96 +1,38 @@
...
-    <PackageReference Include="Microsoft.Azure.Common " Version="2.1.3" />
-    <PackageReference Include="Microsoft.Azure.ResourceManager" Version="1.1.0-preview" />
-  </ItemGroup>
-
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net452' ">
-    <Reference Include="Microsoft.CSharp" />
-    <Reference Include="System.Net" />
-    <Reference Include="System.Net.Http" />
-    <Reference Include="System.Net.Http.WebRequest" />
+    <Compile Remove="Customizations\**" />
```
Comment (shahabhijeet) [inline src/SDKs/Automation/Automation.Tests/Automation.Tests.csproj]: @vrdmr remove this item group completely

File: src/SDKs/Automation/Management.Automation/Microsoft.Azure.Management.Automation.csproj [hunk 4]
```diff
@@ -1,48 +1,25 @@
...
-		<PackageReference Include="System.Collections" Version="4.3.0" />
-		<PackageReference Include="System.Linq" Version="4.3.0" />
-		<PackageReference Include="System.Runtime" Version="4.3.0" />
-		<PackageReference Include="System.Net.Http" Version="4.3.0" />
-	</ItemGroup>
-
-	<!-- Please do not move/edit code below this line -->
-	<Import Condition="Exists('[MSBuild]::GetPathOfFileAbove(AzSdk.RP.props)')" Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
-	<!-- Please do not move/edit code above this line -->
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
```
Comment (shahabhijeet) [inline src/SDKs/Automation/Management.Automation/Microsoft.Azure.Management.Automation.csproj]: @vrdmr this file has not been merged properly. It does not have the upstream changes.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Automation/Automation.Tests/Automation.Tests.csproj (hunk 3 above)
Comment (vrdmr) [reply src/SDKs/Automation/Automation.Tests/Automation.Tests.csproj]: Made the change in the new commit. Pushed in.

File: src/SDKs/Automation/Management.Automation/Microsoft.Azure.Management.Automation.csproj (hunk 4 above)
Comment (vrdmr) [reply src/SDKs/Automation/Management.Automation/Microsoft.Azure.Management.Automation.csproj]: I merged the changes with the upstream. I have changed the csproj to remove any dependencies like the following - 
```
<ItemGroup>
		<PackageReference Include="Hyak.Common" Version="1.1.3" />
		<PackageReference Include="Microsoft.Azure.Common " Version="2.1.3" />
		<PackageReference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework" Version="[1.6.0,2.0.0)" />
	</ItemGroup>
```

The new csproj file removes a lot from existing csproj.

**Comment by shahabhijeet** [comment]
@vrdmr still missing PackageReleaseNotes

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**